CRM/Core/Form/Task/PickProfile.php

```diff
@@ -0,0 +1,160 @@
+   *
+   * @return void
+   */
+  public function preProcess() {
+    // initialize the task and row fields
+    parent::preProcess();
+    $session = CRM_Core_Session::singleton();
+    $this->_userContext = $session->readUserContext();
+
+    CRM_Utils_System::setTitle(ts('Update multiple ' . $this::$entityShortname . 's'));
+
+    // validations
+    if (count($this->_entityIds) > $this->_maxEntities) {
+      CRM_Core_Session::setStatus(ts("The maximum number of %3 you can select for Update multiple %3 is %1. You have selected %2. Please select fewer %3 from your search results and try again.", array(
+        1 => $this->_maxEntities,
+        2 => count($this->_entityIds),
+        3 => $this::$entityShortname . 's',
+      )), ts('Update multiple records error'), 'error');
+      CRM_Utils_System::redirect($this->_userContext);
+    }
+  }
+
+  /**
+   * Build the form object.
+   *
+   *
+   * @return void
+   */
+  public function buildQuickForm() {
+    $types = array(ucfirst($this::$entityShortname));
+    $profiles = CRM_Core_BAO_UFGroup::getProfiles($types, TRUE);
+
+    if (empty($profiles)) {
+      CRM_Core_Session::setStatus(
+        ts("You will need to create a Profile containing the %1 fields you want to edit before you can use Update multiple %2. Navigate to Administer > Customize Data and Screens > Profiles to configure a Profile. Consult the online Administrator documentation for more information.",
+          array(
+            1 => $types[0],
+            2 => $this::$entityShortname . 's',
+          )),
+        ts('Update multiple records error'),
+        'error'
+      );
+      CRM_Utils_System::redirect($this->_userContext);
+    }
+
+    $this->add('select',
+      'uf_group_id',
+      ts('Select Profile'),
+      array(
+        '' => ts('- select profile -'),
+      ) + $profiles,
+      TRUE
+    );
+    $this->addDefaultButtons(ts('Continue'));
+
+    $taskComponent['lc'] = $this::$entityShortname;
+    $taskComponent['ucfirst'] = ucfirst($this::$entityShortname);
+    $this->assign('taskComponent', $taskComponent);
+  }
+
+  /**
+   * Add local and global form rules.
+   *
+   *
+   * @return void
+   */
+  public function addRules() {
+    $this->addFormRule(array('CRM_' . ucfirst($this::$entityShortname) . '_Form_Task_PickProfile', 'formRule'));
+  }
+
+  /**
+   * Global validation rules for the form.
+   *
+   * @param array $fields
+   *   Posted values of the form.
+   *
+   * @return array
+   *   list of errors to be posted back to the form
+   */
+  public static function formRule($fields) {
+    return TRUE;
+  }
+
+  /**
+   * Process the form after the input has been submitted and validated.
+   *
+   *
+   * @return void
+   */
+  public function postProcess() {
+    $params = $this->exportValues();
+
+    $this->set('ufGroupId', $params['uf_group_id']);
+
+    // also reset the batch page so it gets new values from the db
+    $this->controller->resetPage('Batch');
```

Comment:
My IDE tells me the `resetPage` method doesn't exist. Is it just because it's not type-hinted correctly?

Reply:
Yes, I get a few of those (PHPStorm) but I see them elsewhere in the codebase too.  If you have a way to "fix" them let me know, otherwise let's ignore it for now :-)